test/utils/helpers.go

```diff
@@ -376,6 +402,42 @@ func GenerateNameFromText(fullText string, focus []string) string {
 	return strings.ReplaceAll(fullText, " ", "")
 }
 
+func SetAgentConfig(c client.Client, ns *corev1.Namespace, cloudProviders, antreaVersion, kubeconfig string) error {
+	err := c.Create(context.TODO(), ns)
+	if err != nil {
+		return fmt.Errorf("failed to create static vm namespace %+v", err)
+	}
+
+	// TODO: decouple cluster type with cloud provider type
+	clusterType := ""
+	switch cloudProviders {
+	case string(cloudv1alpha1.AzureCloudProvider):
+		clusterType = "aks"
+	case string(cloudv1alpha1.AWSCloudProvider):
+		clusterType = "eks"
+	}
+
+	cmd := exec.Command("./hack/generate-agent-config.sh", "--cluster-type", clusterType, "--antrea-version", antreaVersion)
+	bytes, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to generate antrea agent kubeconfigs %+v: %s", err, string(bytes))
+	}
+	path, err := filepath.Abs("./hack/install-wrapper.sh")
+	if err != nil {
+		return err
+	}
+	dir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	_ = os.Setenv("KUBECONFIG", kubeconfig)
+	_ = os.Setenv("TF_VAR_with_agent", "true")
+	_ = os.Setenv("TF_VAR_antrea_agent_k8s_conf", dir+"/antrea-agent.kubeconfig")
```

Comment:
Reason I dont like this because it introduces dependency between golang code and script. you are expecting script to never change and it should generate file name in hard coded format like antrea-agent.kubeconfig. Rather than this, if you are expecting output from script, pass file location to avoid any dependency. ?

Reply:
I didn't resolve your previous comment on this issue since I'm going to change this but not yet finish